src/ast/mod.rs

```diff
@@ -146,7 +146,38 @@ impl FunctionCtx {
             }
             Call(ref func, ref args) => unimplemented!(),
             Index(ref tup, ref idx) => unimplemented!(),
-            Mktup(ref parts) => unimplemented!(),
+            Mktup(ref parts) => {
+                let reg = self.push_tmp();
+                let mut code = vec![];
+                let mut part_addrs = vec![];
+                let mut start_addr = None;
+                // @TODO: any way to make start_addr not mutable?
+                for (i, part) in parts.iter().enumerate() {
+                    let (part_dest, mut part_code) = self.compile_expr(part);
+                    code.append(&mut part_code);
+                    part_addrs.push(part_dest);
+
+                    if i == 0 {
+                        start_addr = Some(part_dest);
+                    }
+                }
+
+                assert!(start_addr.is_some());
+                let start_addr = start_addr.unwrap();
+                // Minus one required because MkTup is inclusive at the ends,
```

Comment:
Perhaps our bytecode should be `first, len` instead of `first, last`, to allow us to handle empty tuples, and also make this math simpler.